### PR TITLE
drivers: pinmux: stm32l4: Add extra pin functions

### PR DESCRIPTION
**Pull request by gon1332**
Added pins configuration for:
 - PWM2_CH1
 - PWM3_CH2
 - I2C3_{SCL, SDA}
 - SPI2_{SCK, MOSI}

Signed-off-by: Ioannis Konstantelias <ikonstadel@gmail.com>

### CODE REVIEW

**Comment by gon1332** [comment]
recheck

**Comment by gon1332** [comment]
Shippable just timed out. It needs a recheck.

File: drivers/pinmux/stm32/pinmux_stm32l4x.h [hunk 1]
```diff
@@ -52,6 +54,8 @@
 	(STM32_PINMUX_ALT_FUNC_8 | STM32_OPENDRAIN_PULLUP)
 #define STM32L4X_PINMUX_FUNC_PA6_ADC12_IN11 STM32_MODER_ANALOG_MODE
 
+#define STM32L4X_PINMUX_FUNC_PA7_I2C3_SCL                                      \
```
Comment (erwango) [inline drivers/pinmux/stm32/pinmux_stm32l4x.h]: Specific to STM32L496 soc variant. Should be under #ifdef

File: drivers/pinmux/stm32/pinmux_stm32l4x.h [hunk 2]
```diff
@@ -101,13 +105,17 @@
 #define STM32L4X_PINMUX_FUNC_PB3_SPI3_SCK                                      \
 	(STM32_PINMUX_ALT_FUNC_6 | STM32_PUSHPULL_NOPULL)
 
+#define STM32L4X_PINMUX_FUNC_PB4_I2C3_SDA                                      \
```
Comment (erwango) [inline drivers/pinmux/stm32/pinmux_stm32l4x.h]: Specific to STM32L496 soc variant. Should be under #ifdef

**Review by erwango — CHANGES_REQUESTED** [review]

File: drivers/pinmux/stm32/pinmux_stm32l4x.h (hunk 1 above)
Comment (gon1332) [reply drivers/pinmux/stm32/pinmux_stm32l4x.h]: Done.

File: drivers/pinmux/stm32/pinmux_stm32l4x.h (hunk 2 above)
Comment (gon1332) [reply drivers/pinmux/stm32/pinmux_stm32l4x.h]: Done.

**Review by erwango — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**